src/parser.yy

```diff
@@ -181,8 +183,8 @@ void yyerror(bpftrace::Driver &driver, const char *s);
 %%
 
 program:
-                c_definitions config probes_and_subprogs END {
-                    driver.ctx.root = driver.ctx.make_node<ast::Program>($1, $2, std::move($3.second), std::move($3.first), @$);
+                c_definitions config map_decl_list probes_and_subprogs END {
```

Comment:
Can we allow the map_decls to be interspersed with the probes? Seems like you might want the declaration close to usage if you have multiple.

Optionally, before submission I would just have a bug as a "good first issue" and leave a `FIXME(#bug)` here. You don't need this right away, but it's an obvious future improvement.

Reply:
> Can we allow the map_decls to be interspersed with the probes? 

I'd rather not in this iteration (and maybe ever?). I think @danobi and @JakeHillion feel similarly - only for the reason that it's confusing in how it's not really influencing control flow.

Reply:
Ah, I missed this again, sorry! I think my initial implementation did interesperse and I do think that's the slightly more friendly option. As @amscanne says, you might have a map only used in two probes and want to declare it close. But I don't have any issue with this implementation because it's really easy to extend to that behaviour in the future without breaking backwards compatibility.

Reply:
> it's really easy to extend to that behaviour in the future without breaking backwards compatibility

Agreed and we can wait for feedback from users if this is something they find necessary.